tests/test_github.py

```diff
@@ -1789,3 +1795,96 @@ def test_github_enterprise_domain(self, mocker: Any) -> None:
             .add_git_config_key('machete.github.domain', github_enterprise_domain)
         )
         launch_command('github', 'checkout-prs', '--all')
+
```

Comment:
Pls verify that both bugs fixed on #816 would've been captured with these tests (both ordering issue and `hub` config parsing issue)